driver-mapping/src/main/java/com/datastax/driver/mapping/AnnotationParser.java

```diff
@@ -106,18 +112,24 @@ private AnnotationParser() {
         validateOrder(pks, "@PartitionKey");
         validateOrder(ccs, "@ClusteringColumn");
 
-        mapper.addColumns(createColumnMappers(pks, factory, mapper.entityClass, mappingManager, columnCounter),
-                createColumnMappers(ccs, factory, mapper.entityClass, mappingManager, columnCounter),
-                createColumnMappers(rgs, factory, mapper.entityClass, mappingManager, columnCounter));
+        mapper.addColumns(
+                createColumnMappers(pks, factory, mapper.entityClass, mappingManager, columnCounter, tableMetadata, ksName, tableName),
+                createColumnMappers(ccs, factory, mapper.entityClass, mappingManager, columnCounter, tableMetadata, ksName, tableName),
+                createColumnMappers(rgs, factory, mapper.entityClass, mappingManager, columnCounter, tableMetadata, ksName, tableName));
```

Comment:
`ksName` and `tableName` could be retrieved from `tableMetadata` instead of passing them as parameters (also applies to `createFieldMappers` below).